src/main/kotlin/org/jetbrains/teamcity/rest/implementation.kt

```diff
@@ -263,6 +269,20 @@ private class ParameterImpl(private val bean: ParameterBean) : Parameter {
         get() = bean.own
 }
 
+private class ArtifactDependencyImpl(private val bean: ArtifactDependencyBean,
+                                     private val service: TeamCityService) : ArtifactDependency {
+
+    override val disabled: Boolean
+        get() = bean.disabled ?: false
+
+    override val sourceBuildConfiguration: BuildConfiguration
+        get() = BuildConfigurationImpl(bean.`source-buildType`, service)
+
+    override fun fetchProperties(): List<Parameter> = bean.properties
```

Comment:
I think it's better to name function fetchSomething only if it communicates with the server. If you just need to transform values which are already loaded, it would be better to use a different word ('compute'?) or simply convert a function to a property.